Assets/Scripts/Interaction System/Interactable.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Settings;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace InteractionSystem
+{
+    public class Interactable : MonoBehaviour
+    {
+        [SerializeField] private BoxCollider interationArea;
+
+        public UnityEvent onInteraction;
+
+        private bool canBeInteracted;
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.CompareTag("Player"))
+                canBeInteracted = true;
+        }
+
+        private void Update()
+        {
+                  if (canBeInteracted == false)
+                return;
```

Comment:
the indentation is not correct, please clean it, i think you can do it for the whole file by ctrl + shift +o